eeps/eep-0057.md

```diff
@@ -0,0 +1,235 @@
+
+## Triple-quoted strings
+
+Another feature often related to structure documentation is triple-quoted strings. To understand why they may be useful, consider the following example that has a double quote in its documentation:
+
+    -doc "Remove double-quotes (\") from the given string".
+    remove_double_quotes(String) ->
+      ...
+
+For this reason, we may support triple-quoted strings, which reads better in multi-line format and reduces the need for escaping:
+
+    -doc """
+    Remove double-quotes (") from the given string
+    """.
+    remove_double_quotes(String) ->
+      ...
+
+Note, however, this is far from a strict requirement. While Python and Elixir provide this feature, Clojure is a counter example of a language with structured documentation but not triple-quoted strings. However, some features, such as doctests detailed below, may strongly rely on triple-quoted strings.
+
+## Doctests
+
+A direct consequence of making the documentation more structured and accessible is that Erlang can include doctests, which is the ability to run and validate the examples in your documentation. For example, someone could write this:
+
+    -doc """
+    Encodes the given binary to base64.
+
+        1> base64:encode("hello").
+        <<"aGVsbG8=">>
+
+    """.
+    -spec encode(binary()) -> binary().
+    encode(Binary) ->
+      % ....
+
+And then in your test suite:
+
+    doctests(_Config) ->
+      ct_doctest:run(base64).
+
+The doctest attribute will access the documentation entries in the base64 Docs chunk, extract all of the examples, and run them. Of course, there is nothing stopping doctests from being implemented on top of EDoc today, but the fact the documentation is already parsed by the parser/compiler, makes doctests trivial to add. Doctests would benefit from a separate EEP, as there are some extra considerations, as doctesting exceptions, unparseable formats, etc, but it is worth mentioning them given their benefits to users and documentation authors.
+
+EDoc and source code documentation
+==================================
+
+## Should documentation be in the source code?
+
+There is one discussion which is the debate if API Reference documentation should exist in the source code or be placed in a separate file. There are arguments against and in favor of both approaches. If the documentation is alongside the source code, it is easier to keep it up to date. At the same time, it may pollute the source code. This proposal does not advocate that one is better than the other. Rather, because we already have a tool that colocates documentation and source code, namely EDoc, this EEP aims to propose a more integrated variant of said tool.
```

Comment:
```suggestion
There is one discussion which is the debate if API Reference documentation should exist in the source code or be placed in a separate file. There are arguments against and in favor of both approaches. If the documentation is alongside the source code, it is easier to keep it up to date. At the same time, it may pollute the source code. This proposal does not advocate that one is better than the other. Rather, because we already have a tool that colocates documentation and source code, namely EDoc, this EEP aims to propose a more integrated variant of said tool, which will hopefully lead developers to better document their Erlang modules and libraries.
```